argh_derive/src/lib.rs

```diff
@@ -201,6 +200,34 @@ impl<'a> StructField<'a> {
     }
 }
 
+fn to_kebab_case(s: &str) -> String {
+    let words = s.split('_').filter(|word| !word.is_empty());
```

Comment:
:thinking: I am unsure here to be honest: rust already warns if the field name isn't snake case, so the user already wold have to `#[allow(non_snake_case)]`.

So, if we ignore backwrards compat, I am pretty sure that the current behavior is the right one (and probably is better than original Heck's behavior: if I write something like `#[allow(non_snake_case)] XML: bool`, I probably expect `--XML`, not `--xml`). 

But yeah, this is in theory a breaking change.....

I guess, the right thing to do is to emit error, so that we don't get silent breakage, but just remove the extra check in some future relase.

Reply:
Ah, we actually already have validation here, it's just that we wrongly not doing it for implimit names (and, eg, allow non-ascii flags today). PTAL